### PR TITLE
Mesh 3 - avoid tuples in internal code

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

This PR introduces new `struct`s to replace `std::tuple`'s in internal code of Mesh_3, and in particular avoid the use of `std::get<i>(tuple)` that can be hardly readable, compared to the use of explicit member names like `my_tuple_struct.dimension`

## Release Management

* Affected package(s): Mesh_3
* License and copyright ownership: unchanged

### CODE REVIEW

**Comment by janetournois** [comment]
@lrineau what do you think about this internal change?

**Review by lrineau — COMMENTED** [review]
That looks good!

File: Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h [hunk 1]
```diff
@@ -356,6 +360,19 @@ class Refine_facets_3_base
...
+    return std::get<2>(intersection);
+  }
+  Index index(const Intersection& intersection) const
+  {
+    return std::get<1>(intersection);
+  }
+  Point_3 point(const Intersection& intersection) const
+  {
+    return std::get<0>(intersection);
+  }
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h:374]: Here the point is returned by copy. Maybe we could return the ref:

```suggestion
  const Point_3& point(const Intersection& intersection) const
  {
    return std::get<0>(intersection);
  }
```

**Review by lrineau — APPROVED** [review]
Looks good. That is a nice improvement over the existing code.

File: Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h [hunk 2]
```diff
@@ -356,6 +358,19 @@ class Refine_facets_3_base
     return sstr.str();
   }
 
+  const int dimension(const Intersection& intersection) const
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h:361]: The return type `const int` will certainly trigger a warning with one compiler or another. I suggest:

```suggestion
  int dimension(const Intersection& intersection) const
```

File: Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h [hunk 3]
```diff
@@ -356,6 +358,19 @@ class Refine_facets_3_base
...
+  {
+    return std::get<2>(intersection);
+  }
+  const typename MeshDomain::Index& index(const Intersection& intersection) const
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h:365]: Why remove the typedefs for `MeshDomain::Index` and `MeshDomain::Point_3`?

**Comment by sloriot** [comment]
Several errors detected in CGAL-6.0-Ic-271 In Mesh_3, Poisson et Tet remeshing

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.1-Ic-3

**Pull request merged**